File: Tasks/MySolution/Shuffle.java
package Tasks.MySolution;

import java.util.Arrays;

public class Shuffle {
    public static void main(String[] args) {
        Object[] array = { "a", "b", "c", "d", "e" };
        Object[] newArray = shuffleProcess(array);

        String arrayString = Arrays.toString(newArray);
        System.out.println(arrayString);
    }

    public static Object[] shuffleProcess(Object[] arr) {
        Object[] array = arr.clone();
        for (int i = 0; i < array.length; i++) {
            int firstIndex = (int) (Math.random() * array.length);
            int secondIndex = (int) (Math.random() * array.length);
            Object tmp = array[firstIndex];
            array[firstIndex] = array[secondIndex];
            array[secondIndex] = tmp;
        }
        return array;
    }
}

File: refactoring/shuffle.js
// Shuffle an array

'use strict';

const INTERVAL_EQUALIZER = 0.5;
const Shuffle = (List) => {
  const arr = [...List];
  arr.sort(() => Math.random() - INTERVAL_EQUALIZER);
  return arr; 
};

require('../Tests/shuffle.js')(Shuffle);
